back/app/models/static_page.rb

```diff
@@ -20,30 +37,68 @@
 
+  has_many :pins, as: :page, inverse_of: :page, dependent: :destroy
+  has_many :pinned_admin_publications, through: :pins, source: :admin_publication
   has_one :nav_bar_item, dependent: :destroy
   has_many :static_page_files, -> { order(:ordering) }, dependent: :destroy
   has_many :text_images, as: :imageable, dependent: :destroy
   accepts_nested_attributes_for :nav_bar_item
   accepts_nested_attributes_for :text_images
 
+  before_validation :set_code, on: :create
+  before_validation :generate_slug, on: :create
+
+  before_validation :strip_title
+  before_validation :sanitize_top_info_section_multiloc
+  before_validation :sanitize_bottom_info_section_multiloc
+
+  before_destroy :confirm_is_custom, prepend: true
+
   validates :title_multiloc, presence: true, multiloc: { presence: true }
-  validates :body_multiloc, presence: true, multiloc: { presence: true, html: true }
   validates :slug, presence: true, uniqueness: true
   validates :code, inclusion: { in: CODES }
-  validates :code, uniqueness: true, if: ->(page) { !page.custom? }
+  validates :code, uniqueness: true, unless: :custom?
 
-  before_validation :set_code, on: :create
-  before_validation :generate_slug, on: :create
+  validates :banner_enabled, inclusion: [true, false]
+  validates :banner_layout, inclusion: %w[full_width_banner_layout two_column_layout two_row_layout]
+  validates :banner_overlay_color, css_color: true
+  validates :banner_overlay_opacity, numericality: { only_integer: true,
+                                                     in: [0..100],
+                                                     allow_nil: true }
+  validates :banner_header_multiloc, multiloc: true
+  validates :banner_subheader_multiloc, multiloc: true
+  validates :banner_cta_button_type, inclusion: %w[customized_button no_button]
+  with_options if: -> { banner_cta_button_type == 'customized_button' } do
+    validates :banner_cta_button_multiloc, presence: true, multiloc: { presence: true }
+    validates :banner_cta_button_url, presence: true, url: true
+  end
 
-  before_validation :strip_title
-  before_validation :sanitize_body_multiloc
+  validates :top_info_section_enabled, inclusion: [true, false]
+  validates :top_info_section_multiloc, multiloc: { presence: false, html: true }
+
+  validates :projects_enabled, inclusion: [true, false]
+  with_options if: -> { projects_enabled == true } do
+    validates :projects_filter_type, presence: true, inclusion: %w[area topics]
+  end
+
+  validates :events_widget_enabled, inclusion: [true, false]
+
+  validates :bottom_info_section_enabled, inclusion: [true, false]
+  validates :bottom_info_section_multiloc, multiloc: { presence: false, html: true }
+
+  mount_base64_uploader :header_bg, HeaderBgUploader
 
   def custom?
     code == 'custom'
   end
 
   private
 
+  def confirm_is_custom
+    errors.add(:code, 'You cannot delete a page that does not have code: custom') unless custom?
+    throw(:abort) if errors.present?
```

Comment:
The FE should not try to delete this page, so it's a client error, not a server error. So, it's better to return 4xx (400 should be fine) code, not 5xx.

Reply:
I think a 500 on failed delete might be an intentional convention. See my Slack q [here](https://citizenlabco.slack.com/archives/C016C2EHURY/p1661332956597469).

Reply:
If we want to return something other than a 500, how do we get past the fact that it's the `if page.destroyed?` line in the controller that causes the 500, not the else statement?

Reply:
Maybe we could use [this MinimumRequiredValidator](https://github.com/CitizenLabDotCo/citizenlab/blob/9da030bc5af77c89c31693e596cac93b60dda11a/back/app/models/idea_status.rb#L90) used in `IdeaStatus` model?

Reply:
Resolved (see controller #destroy code). Thanks to @alexander-cit for help with this!